modules/LocationUtils.js

```diff
@@ -33,7 +34,19 @@ export const createLocation = (path, state, key, currentLocation) => {
       location.state = state
   }
 
-  location.pathname = decodeURI(location.pathname)
+  try {
+    location.pathname = decodeURI(location.pathname)
+  } catch (e) {
+    if (e instanceof URIError) {
+      warning(
```

Comment:
If we try to create a location with a bad percent-encoding, we swallow the error, keep `location.pathname` encoded, and log a warning.